src/main/java/com/synopsys/integration/detect/tool/binaryscanner/BinaryUploadOperation.java

```diff
@@ -97,4 +106,88 @@ private BinaryUploader createMultipartBinaryScanUploader(File binaryUpload, Name
+            );
+            try {
+                logger.info("Preparing to upload binary scan file: " + binaryScanFile.getAbsolutePath());
+                BinaryScan binaryScan = new BinaryScan(binaryScanFile, projectNameVersion.getName(), projectNameVersion.getVersion(), codeLocationName);
+                BinaryScanBatch binaryScanBatch = new BinaryScanBatch(binaryScan);
+                CodeLocationCreationData<BinaryScanBatchOutput> codeLocationCreationData = binaryScanUploadService.uploadBinaryScan(binaryScanBatch);
+
+                BinaryScanBatchOutput binaryScanBatchOutput = codeLocationCreationData.getOutput();
+                // The throwExceptionForError() in BinaryScanBatchOutput has a bug, so doing that work here
+                throwExceptionForError(binaryScanBatchOutput);
+
+                logger.info("Successfully uploaded binary scan file: " + binaryScanFile.getAbsolutePath());
+                statusEventPublisher.publishStatusSummary(new Status(STATUS_KEY, StatusType.SUCCESS));
+                return codeLocationCreationData;
+            } catch (IntegrationException e) {
+                statusEventPublisher.publishStatusSummary(new Status(STATUS_KEY, StatusType.FAILURE));
+                throw new DetectUserFriendlyException("Failed to upload binary scan file.", e, ExitCodeType.FAILURE_BLACKDUCK_FEATURE_ERROR);
+            }
+        }
+
+        public CodeLocationCreationData<BinaryScanBatchOutput> uploadBinaryScanFiles(
+            BinaryScanBatch binaryScanBatch,
+            BinaryScanUploadService binaryScanUploadService,
+            NameVersion projectNameVersion
+        )
+            throws DetectUserFriendlyException {
+            try {
+                CodeLocationCreationData<BinaryScanBatchOutput> codeLocationCreationData = binaryScanUploadService.uploadBinaryScan(binaryScanBatch);
+
+                BinaryScanBatchOutput binaryScanBatchOutput = codeLocationCreationData.getOutput();
+                // The throwExceptionForError() in BinaryScanBatchOutput has a bug, so doing that work here
+                throwExceptionForError(binaryScanBatchOutput);
+                statusEventPublisher.publishStatusSummary(new Status(STATUS_KEY, StatusType.SUCCESS));
+                return codeLocationCreationData;
+            } catch (IntegrationException e) {
+                statusEventPublisher.publishStatusSummary(new Status(STATUS_KEY, StatusType.FAILURE));
+                throw new DetectUserFriendlyException("Failed to upload binary scan file.", e, ExitCodeType.FAILURE_BLACKDUCK_FEATURE_ERROR);
+            }
+        }
+
+        // BinaryScanBatchOutput used to do this, but our understanding of what needs to happen has been
+        // changing rapidly. Once we're confident we know what it should do, it should presumably move back there.
+        private void throwExceptionForError(BinaryScanBatchOutput binaryScanBatchOutput) throws BlackDuckIntegrationException {
```

Comment:
Thoughts on creating a refactoring ticket? Not majorly important but code with future TODOs tend to live in the code base forever otherwise and if so, we may as well remove this comment

Reply:
Well .. this isn't exactly a future TODO comment and this feature is indeed rapidly evolving so maybe adding another ticket to the backlog isn't the best move.

Reply:
Hmm, maybe anytime we have a doc ticket to update our compatibility matrix we could pair that with a dev ticket to remove legacy code that is for old BD versions that reached end of support? Most of the "check BD version and then decide what to do" code is pretty recent so we'd have to revisit this in a couple years I guess  :|